### PR TITLE
Feature/gsk 1043 debugger tab improvements

### PR DESCRIPTION
**Pull request by kevinmessiaen**
## Type of Change

- [ ] 📚 Examples / docs / tutorials / dependencies update
- [ ] 🔧 Bug fix (non-breaking change which fixes an issue)
- [ ] 🥂 Improvement (non-breaking change which improves an existing feature)
- [ ] 🚀 New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🔐 Security fix

## Checklist

- [ ] I've read the [`CODE_OF_CONDUCT.md`](https://github.com/Giskard-AI/ai-inspector/blob/master/CODE_OF_CONDUCT.md) document.
- [ ] I've read the [`CONTRIBUTING.md`](https://github.com/Giskard-AI/ai-inspector/blob/master/CONTRIBUTING.md) guide.
- [ ] I've updated the code style using `make codestyle`.
- [ ] I've written tests for all new methods and classes that I created.
- [ ] I've written the docstring in Google format for all the methods and classes that I used.

### CODE REVIEW

File: frontend/src/components/AddDebuggingSessionModal.vue [hunk 1]
```diff
@@ -34,20 +42,39 @@ const missingValues = computed(() => {
...
-        });
-
-        closeDialog();
-        emit('createDebuggingSession', newDebuggingSession);
-    } finally {
-        loading.value = false;
+  const mlWorkers = await api.getMLWorkerSettings();
+  const isWorkerAvailable = mlWorkers.find(
+    value => {
+      if (project.value?.mlWorkerType === 'EXTERNAL') {
```
Comment (kevinmessiaen) [inline frontend/src/components/AddDebuggingSessionModal.vue:48]: For me this logic should be in the backend

File: frontend/src/views/main/Main.vue [hunk 2]
```diff
@@ -12,7 +14,9 @@
           </v-list-item>
           <v-divider />
 
-          <v-tooltip v-if="!showProjectTabs" :disabled="showProjectTabs" right>
+          <v-tooltip v-if="projectStore.currentProjectId === null" :disabled="projectStore.currentProjectId !== null" right>
+
+            <!-- <v-tooltip v-if="!showProjectTabs" :disabled="showProjectTabs" right> -->
```
Comment (kevinmessiaen) [inline frontend/src/views/main/Main.vue:19]: We can remove this code

File: frontend/src/views/main/project/modals/SuiteTestInfoModal.vue [hunk 3]
```diff
@@ -74,6 +74,7 @@ import {api} from '@/api';
 import {editor} from 'monaco-editor';
 import TestInputListSelector from "@/components/TestInputListSelector.vue";
 import {useCatalogStore} from "@/stores/catalog";
+import {extractArgumentDocumentation} from "@/utils/python-doc.utils";
```
Comment (kevinmessiaen) [inline frontend/src/views/main/project/modals/SuiteTestInfoModal.vue:77]: Added but not used?

File: frontend/src/views/main/project/TestsCatalog.vue [hunk 4]
```diff
@@ -135,8 +135,9 @@ import StartWorkerInstructions from "@/components/StartWorkerInstructions.vue";
 import { storeToRefs } from "pinia";
 import { useCatalogStore } from "@/stores/catalog";
 import SuiteInputListSelector from "@/components/SuiteInputListSelector.vue";
-import {extractArgumentDocumentation} from "@/utils/python-doc.utils";
-import {alphabeticallySorted} from "@/utils/comparators";
+import { extractArgumentDocumentation } from "@/utils/python-doc.utils";
+import { alphabeticallySorted } from "@/utils/comparators";
+import CodeSnippet from "@/components/CodeSnippet.vue";
```
Comment (kevinmessiaen) [inline frontend/src/views/main/project/TestsCatalog.vue:140]: Added but not used?

File: frontend/src/views/main/project/ProjectsHome.vue [hunk 5]
```diff
@@ -47,7 +47,7 @@
         </v-row>
       </v-card>
       <v-hover v-slot="{ hover }" v-for="p in projects" :key="p.id">
-        <v-card outlined tile class="grey lighten-5 project" :class="[{ 'info': hover }]" :to="{ name: defaultRoute, params: { id: p.id } }" v-show="creatorFilter === 0 || creatorFilter === 1 && p.owner.id === userProfile.id || creatorFilter === 2 && p.owner.id !== userProfile.id">
+        <v-card outlined tile class="grey lighten-5 project" :class="[{ 'info': hover }]" :to="{ name: defaultRoute, params: { id: p.id } }" v-show="creatorFilter === 0 || creatorFilter === 1 && p.owner.id === userProfile.id || creatorFilter === 2 && p.owner.id !== userProfile.id" @click="() => { updateCurrentProject(p.id) }">
```
Comment (kevinmessiaen) [inline frontend/src/views/main/project/ProjectsHome.vue:50]: I think that we shouldn't mix up click and to but add a router push in the updateCurrentProject method to make it easy to maintain

File: frontend/src/views/main/project/modals/SuiteTestInfoModal.vue (hunk 3 above)
Comment (henchaves) [reply frontend/src/views/main/project/modals/SuiteTestInfoModal.vue:77]: It's been used in line 105

File: frontend/src/views/main/project/TestsCatalog.vue (hunk 4 above)
Comment (henchaves) [reply frontend/src/views/main/project/TestsCatalog.vue:140]: Everything is being used here

File: frontend/src/components/AddDebuggingSessionModal.vue (hunk 1 above)
Comment (henchaves) [reply frontend/src/components/AddDebuggingSessionModal.vue:48]: Mapped to https://linear.app/giskard/issue/GSK-1099/disable-actions-that-require-external-worker-if-no-worker-is-connected

Comment (henchaves) [reply frontend/src/components/AddDebuggingSessionModal.vue:48]: Mapped to https://linear.app/giskard/issue/GSK-1099/disable-actions-that-require-external-worker-if-no-worker-is-connected

File: frontend/src/views/main/project/InspectorLauncher.vue [hunk 6]
```diff
@@ -76,13 +88,29 @@ async function launchInspector() {
   mixpanel.track('Create inspection', { datasetId: datasetSelected.value!.id, modelId: props.model.id });
   try {
     creatingInspection.value = true;
+
+    const mlWorkers = await api.getMLWorkerSettings();
```
[the remark this answers is not in the document]
Comment (henchaves) [reply frontend/src/views/main/project/InspectorLauncher.vue:92]: Mapped to https://linear.app/giskard/issue/GSK-1099/disable-actions-that-require-external-worker-if-no-worker-is-connected

**Pull request merged**